src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/RoleAssignmentFilteringResult.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.hmcts.ccd.domain.service.casedataaccesscontrol;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignment;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.matcher.LocationMatcher;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.matcher.RegionMatcher;
+
+import java.util.Map;
+
+@Data
+@AllArgsConstructor
+public class RoleAssignmentFilteringResult {
+    private RoleAssignment roleAssignment;
+    private Map<String, Boolean> filterResults;
+
+    private static boolean isRegionOrLocationFilteringResult(Map.Entry<String, Boolean> mapEntry) {
+        return mapEntry.getKey().equals(RegionMatcher.class.getSimpleName())
+            || mapEntry.getKey().equals(LocationMatcher.class.getSimpleName());
+    }
+
+    public boolean hasPassedFiltering() {
+        if (filterResults.values().isEmpty()) {
+            return false;
```

Comment:
does stream automatically take care of this check?

Reply:
Fixed in https://github.com/hmcts/ccd-data-store-api/pull/1544/commits/19487d5e45f1ee364455c0509f8ee8c3348de2cb

Reply:
sorry.. i was wrong - Stream.allMatch(..) returns true if the stream is empty. So, please double check below method's allMatch logic.

Reply:
Reinstated original code and added test in 8e7b34d